mapie/tests/test_control_risk.py

```diff
@@ -0,0 +1,194 @@
+"""
+Testing for control_risk module.
+Testing for now risks for multilabel classification
+"""
+from typing import Union, List
+
+import pytest
+from mapie._typing import NDArray
+
+import numpy as np
+
+from mapie.control_risk.risks import (_compute_precision,
+                                      _compute_recall)
+
+from mapie.control_risk.p_values import compute_hoefdding_bentkus_p_value
+
+from mapie.control_risk.ltt import (_ltt_procedure,
+                                    _find_lambda_control_star)
+
+
+lambdas = np.array([0.5, 0.9])
+
+y_toy = np.stack([
+    [1, 0, 1],
+    [0, 1, 0],
+    [1, 1, 0],
+    [1, 1, 1],
+])
+
+y_preds_proba = np.stack([
+    [0.2, 0.6, 0.9],
+    [0.8, 0.2, 0.6],
+    [0.4, 0.8, 0.1],
+    [0.6, 0.8, 0.7]
+])
+
+y_preds_proba = np.expand_dims(y_preds_proba, axis=2)
+
+test_recall = np.array([
+    [1/2, 1.],
+    [1., 1.],
+    [1/2, 1.],
+    [0., 1.]
+])
+
+test_precision = np.array([
+    [1/2, 1.],
+    [1., 1.],
+    [0., 1.],
+    [0., 1.]
+])
+
+r_hat = np.array([
+    0.5, 0.8
+])
+
+n = 1100
+
+alpha = np.array([0.6])
+
+valid_index = [[
+    0, 1
+]]
+
+wrong_alpha = 0
+
+wrong_alpha_shape = np.array([
+    [0.1, 0.2],
+    [0.3, 0.4]
+])
+
+wrong_delta = None
+
+
+def test_compute_recall_equal() -> None:
+    """Test that compute_recall give good result"""
+    recall = _compute_recall(lambdas, y_preds_proba, y_toy)
+    np.testing.assert_equal(recall, test_recall)
+
+
+def test_compute_precision() -> None:
+    """Test that compute_precision give good result"""
+    precision = _compute_precision(lambdas, y_preds_proba, y_toy)
+    np.testing.assert_equal(precision, test_precision)
+
+
+def test_recall_with_zero_sum_is_equal_nan() -> None:
+    """Test compute_recall with nan values"""
+    y_toy = np.zeros((4, 3))
+    y_preds_proba = np.random.rand(4, 3, 1)
```

Comment:
UPD: add a random_state to ensure that the test is not modified.

Reply:
To do that, define `random_state = 42` in the beginning of the file.

Reply:
An illustration of what you can do:
```python
random_state = 42
prng = RandomState(random_state)

y_1 = prng.random(51)
y_2 = prng.random((51, 5))
y_3 = prng.randint(0, 2, 51)
```

Reply:
Done

Reply:
```suggestion
    y_preds_proba = prng.rand(4, 3, 1)
```